--- ./src/zephyr/drivers/audio/Kconfig	2021-01-18 12:57:38.339032715 -0800
+++ ./src/zephyr/drivers/audio/Kconfig	2021-01-18 12:58:06.646030027 -0800
@@ -30,6 +30,7 @@
 source "subsys/logging/Kconfig.template.log_config"
 
 source "drivers/audio/Kconfig.tlv320dac"
+source "drivers/audio/Kconfig.cs43l22"
 
 endif # AUDIO_CODEC
 
--- ./src/zephyr/drivers/audio/CMakeLists.txt	2021-01-18 12:58:55.360401049 -0800
+++ ./src/zephyr/drivers/audio/CMakeLists.txt	2021-01-18 12:59:25.395453534 -0800
@@ -3,6 +3,7 @@
 zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_AUDIO_TLV320DAC	tlv320dac310x.c)
+zephyr_library_sources_ifdef(CONFIG_AUDIO_CS43L22	cs43l22.c)
 zephyr_library_sources_ifdef(CONFIG_AUDIO_INTEL_DMIC	intel_dmic.c)
 zephyr_library_sources_ifdef(CONFIG_AUDIO_INTEL_DMIC	decimation/pdm_decim_int32_02_4288_5100_010_095.c)
 zephyr_library_sources_ifdef(CONFIG_AUDIO_INTEL_DMIC	decimation/pdm_decim_int32_02_4375_5100_010_095.c)
--- ./src/zephyr/include/audio/codec.h	2020-09-27 13:15:49.000000000 -0700
+++ ./src/zephyr/include/audio/codec.h	2021-01-19 09:59:08.837085013 -0800
@@ -81,6 +81,14 @@
 	AUDIO_CHANNEL_REAR_CENTER,
 	AUDIO_CHANNEL_SIDE_LEFT,
 	AUDIO_CHANNEL_SIDE_RIGHT,
+	AUDIO_CHANNEL_MASTER_LEFT,
+	AUDIO_CHANNEL_MASTER_RIGHT,
+	AUDIO_CHANNEL_HEADPHONE_LEFT,
+	AUDIO_CHANNEL_HEADPHONE_RIGHT,
+	AUDIO_CHANNEL_SPEAKER_LEFT,
+	AUDIO_CHANNEL_SPEAKER_RIGHT,
+	AUDIO_CHANNEL_PCM_LEFT,
+	AUDIO_CHANNEL_PCM_RIGHT,
 	AUDIO_CHANNEL_ALL,
 } audio_channel_t;
 
--- ./src/zephyr/boards/arm/stm32f4_disco/pinmux.c	2021-01-19 17:23:54.311149048 -0800
+++ ./src/zephyr/boards/arm/stm32f4_disco/pinmux.c	2021-01-19 17:26:22.741841323 -0800
@@ -37,6 +37,10 @@
 	{STM32_PIN_PB5, STM32F4_PINMUX_FUNC_PB5_CAN2_RX},
 	{STM32_PIN_PB13, STM32F4_PINMUX_FUNC_PB13_CAN2_TX},
 #endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(i2c1), okay) && CONFIG_I2C
+	{STM32_PIN_PB6, STM32F4_PINMUX_FUNC_PB6_I2C1_SCL},
+	{STM32_PIN_PB9, STM32F4_PINMUX_FUNC_PB9_I2C1_SDA},
+#endif
 };
 
 static int pinmux_stm32_init(const struct device *port)
--- ./src/zephyr/boards/arm/stm32f4_disco/stm32f4_disco.dts	2020-09-27 13:15:49.000000000 -0700
+++ ./src/zephyr/boards/arm/stm32f4_disco/stm32f4_disco.dts	2021-01-21 19:11:58.538862173 -0800
@@ -57,6 +57,25 @@
 	};
 };
 
+&dma2 {
+	status = "okay";
+};
+
+&i2s1 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	codec0: cs43l22@4a {
+		compatible = "cirrus,cs43l22";
+		reg = <0x4a>;
+		label = "CS43L22";
+		reset-gpios = <&gpiod 4 0>;
+	};
+};
+
 &usart1 {
 	current-speed = <115200>;
 	status = "okay";
